syntax = "proto3";

import "google/protobuf/any.proto";

message Elimination {
  fixed64  id = 1; // 淘汰ID，为ItemID或者UserID
  int32 etime = 2; // 淘汰时间
}

// ID集合
message IdSet {
  repeated fixed64 id = 1 [packed=true];
}

message LogRecord {
  google.protobuf.Any record = 1;
}

message LogAction {
  fixed64       user_id = 1;
  fixed64       item_id = 2; // 点击的item_id
  int32          action = 3; // 用户行为
  int32     action_time = 4; // 用户操作时间
  string dislike_reason = 5; // 不喜欢原因
}

message KeyStr {
  fixed64 key = 1;
  string  str = 2;
}

message LogSubscribe {
  fixed64        user_id = 1;
  repeated KeyStr keystr = 2; // 订阅ID
}

message KeyTime {
  fixed64 key = 1;
  int32 ctime = 2;
}

message LogUserInfo {
  fixed64              user_id = 1;
  int32                  ctime = 2; // 用户最后活跃时间
  repeated KeyStr    subscribe = 3;
  repeated KeyStr      dislike = 4;
  repeated KeyTime      readed = 5;
  repeated KeyTime recommended = 6;
}

message LogCandidateSet {
  fixed64                        user_id = 1;
  repeated fixed64 cand_id = 2 [packed=true];
}

message LogItemInfo {
  fixed64             item_id = 1;
  float                 power = 2; // 初选权重
  int32	         publish_time = 3; // 发布时间
  int32             item_type = 4; // item类型(0 咨询，1 视频)
  int32           picture_num = 5; // 图片个数
  int32           click_count = 6;
  int32            click_time = 7;
  int32           category_id = 8; // 分类ID
  repeated KeyStr   region_id = 9; // 所属地区
  repeated KeyStr belongs_to = 10; 
}


